src/logic/wallets/ethTransactions.ts

```diff
@@ -38,18 +41,14 @@ export const calculateGasPrice = async (): Promise<string> => {
   return Promise.reject(err)
 }
 
-export const calculateGasOf = async (txConfig: {
-  to: string
-  from: string
-  data: string
-  gasPrice?: number
-  gas?: number
-}): Promise<number> => {
-  const web3 = getWeb3()
+export const calculateGasOf = async (txConfig: EthAdapterTransaction): Promise<number> => {
   try {
-    const gas = await web3.eth.estimateGas(txConfig)
+    const ethAdapter = new Web3Adapter({
+      web3: getWeb3(),
+      signerAddress: ZERO_ADDRESS,
```

Comment:
Can we add a comment that a correct signer address is not essential because this is a read-only method?